File: apps/backend/TecoApi/Models/Entities/Order.cs
using TecoApi.Models.Enums;
namespace TecoApi.Models.Entities;

public class Order
{
    public long Id { get; set; }
    public long ProposalId { get; set; }
    public Proposal? Proposal { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.PENDING;
    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.HELD;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }
    public bool ClientConfirmed { get; set; } = false;

    public Review? Review { get; set; }
}

File: app/types/order.ts
import Review from "@/types/review";

interface Order {
    id: number;
    proposalId: number;
    status: OrderStatus;
    paymentStatus: PaymentStatus;
    totalPrice: number;
    createdAt: Date;
    startedAt: Date;
    finishedAt: Date;
    clientConfirmed: boolean;
    review?: Review
}

enum OrderStatus {
    Pending = "PENDING",
    Starting = "STARTING",
    Completed = "COMPLETED",
    Cancelled = "CANCELLED"
}

enum PaymentStatus {
    Held = "HELD",
    Released = "RELEASED",
    Refunded = "REFUNDED"
}

export default Order;